src/OMSimulatorPython/system.py

```diff
@@ -202,6 +202,34 @@ def addSSV(self, cref: CRef, resource: str):
         raise ValueError(f"Component '{first}' not found in {self.name}")
       self.elements[first].addSSV(resource)
 
+  def split_cref(self, cref: CRef):
+    if cref.is_root():
+      return ("", str(cref))
+    else:
+      return (cref.first(), cref.last())
+
+  def _addConnection(self, cref1: CRef, cref2: CRef):
+    first1 = cref1.first()
+    first2 = cref2.first()
+
+    # Both CRefs start in the same subsystem
+    if first1 == first2:
+      if first1 not in self.elements:
+        raise ValueError(f"System '{first1}' not found in '{self.name}'")
+      if first2 not in self.elements:
+        raise ValueError(f"System '{first2}' not found in '{self.name}'")
+
+      (start_element, start_connector) = self.split_cref(cref1.pop_first())
+      (end_element, end_connector) = self.split_cref(cref2.pop_first())
+      # Add the connection inside the subsystem
+      self.elements[first1].addConnection(start_element, start_connector, end_element, end_connector)
```

Comment:
This will not work if the connection should be put into a deeper-nested sub-system. This function should be recursive.